bears/general/VariableCasingBear.py

```diff
@@ -0,0 +1,143 @@
+            special_chars = coalang_keys["special_chars"]
+            delim = re.escape(str(special_chars) + " \t\n")
+            in_quotes = False
+            result_texts = []
+            text_changes = {}
+            lines_changed = set()
+            for line_number, line in enumerate(file, start=1):
+                split = re.split("[" + delim + "]", line)
+                for s in split:
+                    if s.startswith("\""):
+                        in_quotes = True
+                        s = s[1:]
+                    if in_quotes:
+                        if s[-1:] == "\"" and s[-2:] != "\\\"":
+                            in_quotes = False
+                        continue
+                    if len(s) == 0 or s[0].isdigit():
+                        continue
+                    if len(ignore_globs) > 0 and fnmatch(s, ignore_globs):
+                        continue
+                    if s not in keywords:
+                        rep = convertor(s)
+                        if s != rep:
+                            text_changes[s] = rep
+            if len(text_changes) > 0:
+                for prev, new in text_changes.items():
+                    diff = Diff(file)
+                    first_line = -1
+                    num_changes = 0
+                    skip_var = False
+                    for line_number, line in enumerate(file, start=1):
+                        rep = re.sub("(?P<g1>[\\" + delim + "])" + prev +
+                                     "(?P<g2>[\\" + delim + "])",
+                                     "\g<g1>" + new + "\g<g2>", line)
+                        if rep != line:
+                            if line_number in lines_changed:
+                                skip_var = True
+                            lines_changed.add(line_number)
+                            diff.change_line(line_number, line, rep)
+                            num_changes += 1
+                            if first_line == -1:
+                                first_line = line_number
+                    if skip_var:
+                        continue
+                    msg = "Change '" + prev + "' to '" + new + "'"
+                    if num_changes > 1:
+                        msg += ": " + str(num_changes) + " lines affected"
+                    result_texts.append(msg)
+                    vio = "The following name changes are suggested:"
+                    vio += "".join("\n- " + string
+                                   for string in result_texts)
+                    yield Result.from_values(
+                        self,
+                        vio,
+                        diffs={filename: diff},
+                        file=filename,
+                        line=first_line)
+                    result_texts = []
+        else:
+            self.warn("keywords and special_chars not defined for", language)
+            self.warn("Not running VariableCasingBear")
+
+    def all_to_camel(self, line):
```

Comment:
`line` is I think a bad name (it's not the line number and not the complete line, but the string you want to convert to camelcase) + you don't need self + what is `all`?